src/systemctl/systemctl.c

```diff
@@ -7865,6 +7865,14 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
                                 return log_oom();
                         break;
 
+                case '.':
+                        /* Output an error mimicking getopt, and print a hint afterwards */
+                        log_error("%s: invalid option -- '.'", program_invocation_name);
+                        log_notice("Hint: to specify units starting with a dash, use \"--\":\n"
+                                   "      %s [OPTIONS...] {COMMAND} -- -.%s ...",
```

Comment:
maybe add "Example:" to this line to clarify that this is an example?

Reply:
It's actually a bit more than an example, because it uses `optarg` to construct a part of the correct commandline. Alos, there's an ellipsis at the end, so I think it's pretty clear that this is not the full thing.